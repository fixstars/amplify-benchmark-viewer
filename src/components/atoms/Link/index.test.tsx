/**
 * Copyright (c) Fixstars Corporation and Fixstars Amplify Corporation.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { render, screen, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { BrowserRouter, Router } from 'react-router-dom'

import { Link } from '.'

describe('<Link />', () => {
  it('Rendered well', async () => {
    const { container } = render(
      <BrowserRouter>
        <Link to="/pages/1">test link</Link>
      </BrowserRouter>,
    )

    const link = screen.getByText('test link')
    expect(link).toHaveStyle(`
      text-decoration: none;
      color: initial;
    `)

    expect(container).toMatchSnapshot()
  })

  it('Go to the link URL when the link is clicked', () => {
    const mockPush = jest.fn()
    const history = createMemoryHistory({ initialEntries: ['/'] })
    history.push = mockPush

    render(
      <Router location={history.location} navigator={history}>
        <Link to="/pages/1">test link</Link>
      </Router>,
    )

    fireEvent.click(screen.getByText('test link'))
    expect(mockPush.mock.calls[0][0].pathname).toBe('/pages/1')
  })
})
