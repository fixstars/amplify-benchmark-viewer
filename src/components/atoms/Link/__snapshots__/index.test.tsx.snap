// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Rendered well 1`] = `
<div>
  <a
    class="css-94l1l1"
    href="/pages/1"
  >
    test link
  </a>
</div>
`;
